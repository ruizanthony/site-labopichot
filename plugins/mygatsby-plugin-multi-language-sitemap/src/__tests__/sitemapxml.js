import {
  wrapWithXMLHeader,
  wrapWithSitemapIndex,
  wrapWithSitemapAndLoc,
  wrapWithSitemapAndLocs,
  wrapWithUrlset,
  wrapWithUrl,
  wrapWithLoc,
  wrapWithXMLLink
} from '../sitemapxml'

describe(`sitemapxml`, () => {
  const str = `https://www.automizely.com/assets/v3/sitemap-0.xml`
  const strs = [
    `https://www.automizely.com/assets/v3/sitemap-0.xml`, 
    `https://www.automizely.com/assets/v3/sitemap-1.xml`,
    `https://www.automizely.com/assets/v3/sitemap-2.xml`
  ]
  it(`wrapWithSitemapAndLoc`, () => {
    const result = wrapWithSitemapAndLoc(str)
    expect(result).toMatchSnapshot()
  })
  it(`wrapWithSitemapAndLocs`, () => {
    const result = wrapWithSitemapAndLocs(strs)
    expect(result).toMatchSnapshot()
  })
  it(`wrapWithSitemapIndex`, () => {
    const result = wrapWithSitemapIndex(str)
    expect(result).toMatchSnapshot()
  })
  it(`wrapWithXMLHeader`, () => {
    const result = wrapWithXMLHeader(str)
    expect(result).toMatchSnapshot()
  })
  it(`wrapWithUrlset`, () => {
    const result = wrapWithUrlset(str)
    expect(result).toMatchSnapshot()
  })
  it(`wrapWithUrl`, () => { 
    const result = wrapWithUrl(str)
    expect(result).toMatchSnapshot()
  })
  it(`wrapWithLoc`, () => {
    const result = wrapWithLoc(str)
    expect(result).toMatchSnapshot() 
  })
  it(`wrapWithXMLLink`, () => {
    const result = wrapWithXMLLink(str)
    expect(result).toMatchSnapshot()  
  })
})