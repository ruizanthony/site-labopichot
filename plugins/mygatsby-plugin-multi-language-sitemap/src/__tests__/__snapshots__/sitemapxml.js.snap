// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sitemapxml wrapWithLoc 1`] = `"<loc>https://www.automizely.com/assets/v3/sitemap-0.xml</loc>"`;

exports[`sitemapxml wrapWithSitemapAndLoc 1`] = `"<sitemap><loc>https://www.automizely.com/assets/v3/sitemap-0.xml</loc></sitemap>"`;

exports[`sitemapxml wrapWithSitemapAndLocs 1`] = `"<sitemap><loc>https://www.automizely.com/assets/v3/sitemap-0.xml</loc></sitemap><sitemap><loc>https://www.automizely.com/assets/v3/sitemap-1.xml</loc></sitemap><sitemap><loc>https://www.automizely.com/assets/v3/sitemap-2.xml</loc></sitemap>"`;

exports[`sitemapxml wrapWithSitemapIndex 1`] = `"<sitemapindex xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\">https://www.automizely.com/assets/v3/sitemap-0.xml</sitemapindex>"`;

exports[`sitemapxml wrapWithUrl 1`] = `"<url>https://www.automizely.com/assets/v3/sitemap-0.xml</url>"`;

exports[`sitemapxml wrapWithUrlset 1`] = `"<urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:xhtml=\\"http://www.w3.org/1999/xhtml\\">https://www.automizely.com/assets/v3/sitemap-0.xml</urlset>"`;

exports[`sitemapxml wrapWithXMLHeader 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>https://www.automizely.com/assets/v3/sitemap-0.xml"`;

exports[`sitemapxml wrapWithXMLLink 1`] = `"<xhtml:link rel=\\"alternate\\" hreflang=\\"https://www.automizely.com/assets/v3/sitemap-0.xml\\" href=\\"undefined\\" />"`;
